api/src/services/session.service.ts

```diff
@@ -0,0 +1,142 @@
+}
+
+export class SessionService {
+  private logger: FastifyBaseLogger;
+  private cdpService: CDPService;
+  private seleniumService: SeleniumService;
+  public activeSession: SessionDetails;
+
+  constructor(config: { cdpService: CDPService, seleniumService: SeleniumService, logger: FastifyBaseLogger }) {
+    this.cdpService = config.cdpService;
+    this.seleniumService = config.seleniumService;
+    this.logger = config.logger;
+    this.activeSession = {
+      id: uuidv4(),
+      createdAt: new Date().toISOString(),
+      ...defaultSession,
+      completion: Promise.resolve(),
+      complete: () => {},
+    }
+  }
+
+  public async startSession(options: {
+    sessionId?: string;
+    proxyUrl?: string;
+    userAgent?: string;
+    sessionContext?: Record<string, any>;
+    isSelenium?: boolean;
+    logSinkUrl?: string;
+    blockAds?: boolean;
+    extensions?: string[];
+    timezone?: string;
+    dimensions?: { width: number; height: number };
+  }): Promise<SessionDetails> {
+    const {
+      sessionId,
+      proxyUrl,
+      userAgent,
+      sessionContext,
+      extensions,
+      logSinkUrl,
+      timezone,
+      dimensions,
+      isSelenium,
+      blockAds,
+    } = options;
+
+    const browserLauncherOptions: BrowserLauncherOptions = {
+      options: {
+        headless: true,
+        args: [userAgent ? `--user-agent=${userAgent}` : undefined].filter(Boolean) as string[],
+        proxyUrl,
+      },
+      cookies: sessionContext?.cookies || [],
+      userAgent: sessionContext?.userAgent,
+      blockAds,
+      extensions: extensions || [],
+      logSinkUrl,
+      timezone: timezone || "US/Pacific",
+      dimensions,
+    };
+
+    if (isSelenium) {
+      await this.cdpService.shutdown();
+      await this.seleniumService.launch(browserLauncherOptions);
+      
+      return this.resetSessionInfo({
+        id: sessionId || uuidv4(),
+        status: "live",
+        websocketUrl: "",
+        debugUrl: "",
+        sessionViewerUrl: "",
+        userAgent:
+          sessionContext?.userAgent ||
+          "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36",
+        proxy: proxyUrl,
+        solveCaptcha: false,
+        isSelenium,
+      });
+    } else {
+      await this.cdpService.startNewSession(browserLauncherOptions);
+      return this.resetSessionInfo({
+        id: sessionId || uuidv4(),
+        status: "live",
+        websocketUrl: `ws://${env.DOMAIN ?? env.HOST}:${env.PORT}/`,
+        debugUrl: `http://${env.DOMAIN ?? env.HOST}:${env.PORT}/v1/devtools/inspector.html`,
+        sessionViewerUrl: `http://${env.DOMAIN ?? env.HOST}:${env.PORT}`,
+        userAgent: this.cdpService.getUserAgent(),
+        proxy: proxyUrl,
+        solveCaptcha: false,
+        isSelenium,
+      });
+    }
+  }
+
+  public async endSession(): Promise<void> {
+    this.activeSession.complete();
+    this.activeSession.status = "released";
```

Comment:
```suggestion
    this.activeSession.status = "pending";
```

So I think the status should be set back to "pending" after a session is released right -- since the default session is relaunched in the `cdpService.endSession()` [here](https://github.com/steel-dev/steel-browser/blob/62af6d711e13e39f87d55d59f9f3f98e6e41933d/api/src/services/cdp.service.ts#L511):

Reply:
Or perhaps after the below if statement

Reply:
Yeah, we actually need to reset all session info after that if statement. I added a call to `resetSessionInfo()` after it, so it will become a new `pending` session then